plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -693,7 +698,6 @@ void state_history_plugin::plugin_startup() {
    handle_sighup(); // setup logging
 
    try {
-      my->thr = std::thread([ptr = my.get()] { ptr->ctx.run(); });
       my->listen();
```

Comment:
Seems like `listen()` should be posted to the ship thread.

Reply:
If we do post `listen()` to the ship thread, we need either https://github.com/AntelopeIO/leap/pull/632 or something specific only for this use.

Reply:
It's too bad you can't just start the thread after calling `my->listen()`. That seems like a rather unfortunate behavior of `named_thread_pool` that it blocks that usage pattern.

Reply:
I don't follow. How would that make any difference?

Reply:
I see what you mean. I find it easier to reason about the exclusivity by posting it to the ship thread. I agree it is nice to acquire resources in startup as it simplifies error handling and error reporting. 

For this PR, I modified `named_thread_pool` to allow for delay start of the threads and changed to start the thread after the call to `listen()`.

I would like to get https://github.com/AntelopeIO/leap/pull/632 into `main` along with this PRs delay start of `named_thread_pool`.